src/main/java/jp/co/yahoo/yosegi/tools/FindBestSortKey.java

```diff
@@ -0,0 +1,333 @@
+      this.realDataSize = realDataSize;
+    }
+
+    @Override
+    public int compare( final KeyAndRealDataSize k1 , final KeyAndRealDataSize k2 ) {
+      return Long.compare( k1.realDataSize , k2.realDataSize );
+    }
+
+    @Override
+    public int compareTo( final KeyAndRealDataSize target ) {
+      return realDataSize.compareTo( target.realDataSize );
+    }
+
+    @Override
+    public String toString() {
+      Map<Object,Object> dataSet = new HashMap<Object,Object>();
+      dataSet.put( "keys" , keys );
+      dataSet.put( "real_data_size" , realDataSize );
+      try {
+        return new String( new JacksonMessageWriter().create( dataSet ) );
+      } catch ( IOException ex ) {
+        throw new RuntimeException( ex );
+      }
+    }
+
+  }
+
+  public static Options createOptions( final String[] args ){
+
+    Option input = OptionBuilder.
+      withLongOpt("input").
+      withDescription("Input file path.").
+      hasArg().
+      isRequired().
+      withArgName("input").
+      create( 'i' );
+
+    Option output = OptionBuilder.
+      withLongOpt("output").
+      withDescription("output and work directory path.").
+      hasArg().
+      isRequired().
+      withArgName("output").
+      create( 'o' );
+
+    Option lines = OptionBuilder.
+      withLongOpt("lines").
+      withDescription("Number of input lines.").
+      hasArg().
+      withArgName("lines").
+      create( 'n' );
+
+    Option columns = OptionBuilder.
+      withLongOpt("columns").
+      withDescription("Number of colums.").
+      hasArg().
+      withArgName("columns").
+      create( 'c' );
+
+    Option help = OptionBuilder.
+      withLongOpt("help").
+      withDescription("help").
+      withArgName("help").
+      create( 'h' );
+
+    Options  options = new Options();
+
+    return options
+      .addOption( input )
+      .addOption( output )
+      .addOption( lines )
+      .addOption( columns )
+      .addOption( help );
+  }
+
+  public static void printHelp( final String[] args ){
+    HelpFormatter hf = new HelpFormatter();
+    hf.printHelp( "[options]" , createOptions( args ) );
+  }
+
+  public static void createOriginalFile(
+      final String input , final String originalFile , final long lines ) throws IOException {
+    Configuration config = new Configuration();
+
+    InputStream in = FileUtil.fopen( input );
+    YosegiSchemaReader reader = new YosegiSchemaReader();
+    File file = new File( input );
+    long fileLength = file.length();
+    reader.setNewStream( in , fileLength , config );
+
+    OutputStream out = FileUtil.create( originalFile );
+    IStreamWriter writer = StreamWriterFactory.create( out , "json" , null );
+
+    long inCount = 0;
+    long skipCount = 0;
+    while( reader.hasNext() && inCount < lines ){
+      if( skipCount < 10000 ) {
+        skipCount++;
+        continue;
+      }
+      writer.write( reader.next() );
+      inCount++;
+    }
+    writer.close();
+    reader.close();
```

Comment:
I think "try-with-resources" statement uses for the resources such as reader and writer that need close operation. I don't know YosegiSchemaReader can write using this statement.However, I think IStreamWriter can use it.
At close operation for closable object has a risk of memory leak, so, please consider to use "try-with-resources" statement.

Reply:
I think that's true.
But first, we need to make the Closeable implementation complete with writer and reader.